bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java

```diff
@@ -1563,6 +1601,55 @@ public void handleV3Response(BookkeeperProtocol.Response response) {
         }
     }
 
+    class ForceLedgerCompletion extends CompletionValue {
+        final ForceLedgerCallback cb;
+
+        public ForceLedgerCompletion(final CompletionKey key,
+                                  final ForceLedgerCallback originalCallback,
+                                  final Object originalCtx,
+                                  final long ledgerId) {
+            super("ForceLedger",
+                  originalCtx, ledgerId, BookieProtocol.LAST_ADD_CONFIRMED,
+                  forceLedgerOpLogger, forceLedgerTimeoutOpLogger);
+            this.cb = new ForceLedgerCallback() {
+                    @Override
+                    public void forceLedgerComplete(int rc, long ledgerId,
+                                                 BookieSocketAddress addr,
+                                                 Object ctx) {
+                        logOpResult(rc);
+                        originalCallback.forceLedgerComplete(rc, ledgerId,
+                                                          addr, originalCtx);
+                        key.release();
+                    }
+                };
+        }
+
+        @Override
+        public void errorOut() {
+            errorOut(BKException.Code.BookieHandleNotAvailableException);
+        }
+
+        @Override
+        public void errorOut(final int rc) {
+            errorOutAndRunCallback(
+                    () -> cb.forceLedgerComplete(rc, ledgerId, addr, ctx));
+        }
+
+        @Override
+        public void handleV3Response(BookkeeperProtocol.Response response) {
+            ForceLedgerResponse forceLedgerResponse = response.getForceLedgerResponse();
+            StatusCode status = response.getStatus() == StatusCode.EOK
+                ? forceLedgerResponse.getStatus() : response.getStatus();
+            long ledgerId = forceLedgerResponse.getLedgerId();
+
+            if (LOG.isDebugEnabled()) {
+                logResponse(status, "ledger", ledgerId);
```

Comment:
seems `logResponse` already has an `isDebugEnabled()` internal.

Reply:
this is consistent with all the rest of the file, I can change every where, but I think that it can be a slight save of resources.
I would prefer to keep the style consistent.
I will change if you feel strong about it